src/main.rs

```diff
@@ -1,44 +1,19 @@
-mod env;
-mod error;
-mod handlers;
-mod middleware;
-mod providers;
-mod relay;
-mod state;
-mod stores;
-
-use crate::state::{Metrics, TenantStoreArc};
-use crate::stores::tenant::DefaultTenantStore;
-use axum::{
-    routing::{delete, get, post},
-    Router,
-};
 use dotenv::dotenv;
-use opentelemetry::sdk::metrics::selectors;
-use opentelemetry::sdk::{
-    trace::{self, IdGenerator, Sampler},
-    Resource,
-};
-use opentelemetry::util::tokio_interval_stream;
-use opentelemetry::KeyValue;
-use opentelemetry_otlp::{Protocol, WithExportConfig};
+use echo_server::error::Result;
+use echo_server::state::TenantStoreArc;
+use echo_server::stores::tenant::DefaultTenantStore;
 use sqlx::postgres::{PgConnectOptions, PgPoolOptions};
 use sqlx::ConnectOptions;
-use std::net::SocketAddr;
 use std::str::FromStr;
 use std::sync::Arc;
 use std::time::Duration;
-use tower::ServiceBuilder;
-use tower_http::trace::{DefaultMakeSpan, DefaultOnRequest, DefaultOnResponse, TraceLayer};
 use tracing::log::LevelFilter;
-use tracing::{warn, Level};
-use tracing_subscriber::fmt::format::FmtSpan;
 
 #[tokio::main]
-async fn main() -> error::Result<()> {
+async fn main() -> Result<()> {
     dotenv().ok();
-    let config =
-        env::get_config().expect("Failed to load config, please ensure all env vars are defined.");
+    let config = echo_server::env::get_config()
+        .expect("Failed to load config, please ensure all env vars are defined.");
 
     let mut supported_providers_string = "multi-tenant".to_string();
```

Comment:
Move inside of bootstrap function

Reply:
No I need to be able to not use `env` when bootstrapping from functional test

Reply:
`env::get_config()` just provides a `Config` which is what this string is generated from. If you are mocking a `Config` then it will mock this variable too.